jenkins-jobs/job-dsl/mongodb_matrix_test.groovy

```diff
@@ -44,8 +43,25 @@ matrixJob('debezium-mongodb-matrix-test') {
     }
 
     steps {
-        maven {
-            goals('clean install -U -s $HOME/.m2/settings-snapshots.xml -pl debezium-connector-mongodb -am -fae -Dmaven.test.failure.ignore=true -Dversion.mongo.server=$MONGODB_VERSION')
-        }
+        shell('''
+# Ensure WS cleaup
+ls -A1 | xargs rm -rf
+
+# Retrieve sources
+if [ "$PRODUCT_BUILD" == true ] ; then
+    PROFILE_PROD="pnc"
+    curl -OJs $SOURCE_URL && unzip debezium-*-src.zip
+else
+    PROFILE_PROD="none"
+    git clone $REPOSITORY . 
+    git checkout $BRANCH
+fi
+                    
+# Run maven build
+mvn clean install -U -s $HOME/.m2/settings-snapshots.xml -pl debezium-connector-mongodb -am -fae \
+    -Dmaven.test.failure.ignore=true \
+    -Dversion.mongo.server=$MONGODB_VERSION \
+    -P$PROFILE_PROD 
```

Comment:
@jcechace  What are `pnc` and `none` profiles?

Reply:
pnc is a profile set in settings.xml (available in our jenkins) containing indy repository and plugin repository. None is a non-exsistent profile (ignored by maven) used to set the variable if the build is not product-based.